File: src/main/java/io/scriptor/type/ChannelResultConverter.java
package io.scriptor.type;

import io.scriptor.http.result.HTTPResult;
import io.scriptor.http.result.HTTPResultChannel;
import io.scriptor.http.result.HTTPResultVoid;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.nio.channels.ReadableByteChannel;

public class ChannelResultConverter implements IConverter<ReadableByteChannel, HTTPResult<?>> {

    @Override
    public @NotNull HTTPResult<?> from(final @Nullable ReadableByteChannel source) {
        if (source == null) {
            return new HTTPResultVoid(200, "OK");
        }
        return new HTTPResultChannel(200, "OK", source);
    }
}

File: src/main/java/io/scriptor/type/ChannelResultConverter.kt
package io.scriptor.type

import io.scriptor.http.result.HTTPResult
import io.scriptor.http.result.HTTPResultChannel
import io.scriptor.http.result.HTTPResultVoid
import java.nio.channels.ReadableByteChannel

class ChannelResultConverter : IConverter<ReadableByteChannel, HTTPResult<*>> {

    override fun from(source: ReadableByteChannel?): HTTPResult<*> {
        if (source == null) {
            return HTTPResultVoid(200, "OK")
        }
        return HTTPResultChannel(200, "OK", source)
    }
}
